Clarify how a boolean query's must_not works with multiple clauses (#9304)
opensearch-project/documentation-website: _query-dsl/compound/bool.md

Clause | Behavior
:--- | :---
`must` | Logical `and` operator. The results must match all queries in this clause. 

> `must_not` | Logical `not` operator. All matches are excluded from the results. If `must_not` has multiple clauses, only documents which match none of those clauses will be returned. For example, `"must_not":[{clause_A}, {clause_B}]` is equivalent to `NOT(A OR B)`. 

Reviewer: ```suggestion
`must_not` | Logical `not` operator. All matches are excluded from the results. If `must_not` has multiple clauses, only documents that do not match any of those clauses are returned. For example, `"must_not":[{clause_A}, {clause_B}]` is equivalent to `NOT(A OR B)`. 
```
